python/lsst/ip/isr/crosstalk.py

```diff
@@ -163,15 +169,31 @@ def fromDict(cls, dictionary):
                                f"found {dictionary['metadata']['OBSTYPE']}")
 
         calib.setMetadata(dictionary['metadata'])
-        calib._detectorName = dictionary['metadata']['DETECTOR']
-        calib._detectorSerial = dictionary['metadata']['DETECTOR_SERIAL']
+        calib._detectorName = dictionary.get('detectorName',
+                                             dictionary['metadata'].get('DETECTOR', None))
+        calib._detectorSerial = dictionary.get('detectorSerial',
+                                               dictionary['metadata'].get('DET_SER', None))
+        calib._instrument = dictionary.get('instrument',
+                                           dictionary['metadata'].get('INSTRUME', None))
 
         calib.hasCrosstalk = dictionary.get('hasCrosstalk',
                                             dictionary['metadata'].get('HAS_CROSSTALK', False))
         if calib.hasCrosstalk:
             calib.nAmp = dictionary.get('nAmp', dictionary['metadata'].get('NAMP', 0))
             calib.crosstalkShape = (calib.nAmp, calib.nAmp)
             calib.coeffs = np.array(dictionary['coeffs']).reshape(calib.crosstalkShape)
+            if 'coeffErr' in dictionary:
+                calib.coeffErr = np.array(dictionary['coeffErr']).reshape(calib.crosstalkShape)
+            else:
+                calib.coeffErr = np.zeros_like(calib.coeffs)
+            if 'coeffNum' in dictionary:
+                calib.coeffNum = np.array(dictionary['coeffNum']).reshape(calib.crosstalkShape)
+            else:
+                calib.coeffNum = np.zeros_like(calib.coeffs, dtype=int)
+            if 'coeffValid' in dictionary:
+                calib.coeffValid = np.array(dictionary['coeffValid']).reshape(calib.crosstalkShape)
+            else:
+                calib.coeffValid = np.zeros_like(calib.coeffs, dtype=bool)
```

Comment:
I read this as "If it's not there, we make them all invalid" - is that right, or should this be `np.ones_like()`?

Reply:
The other two should be zero, and I think I just copied too much.